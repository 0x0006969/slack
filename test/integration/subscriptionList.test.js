const request = require('supertest');
const nock = require('nock');

const helper = require('.');
const fixtures = require('../fixtures');

const { probot } = helper;
// create a bunch of subscriptions in database
// receive slash command for subscription list
// nock assertions

describe('Integration: subscription list', async () => {
  const { SlackWorkspace, Installation, Subscription } = helper.robot.models;

  const workspace = await SlackWorkspace.create({
    slackId: 1,
    accessToken: 'secret',
  });

  const installation = await Installation.create({
    githubId: 1,
    ownerId: 1,
  });

  await Subscription.create({
    githubId: 1,
    channelId: 'C2147483705',
    installationId: installation.id,
    slackWorkspaceId: workspace.id,
  });

  await Subscription.create({
    githubId: 2,
    channelId: 'C2147483705',
    installationId: installation.id,
    slackWorkspaceId: workspace.id,
  });

  test('works', async () => {
    const command = fixtures.slack.command({
      text: 'subscribe list',
    });

    await request(probot.server).post('/slack/command').send(command)
     .expect(200)
     .expect((res) => {
       expect(res.body).toMatchSnapshot();
     });
  });
});
