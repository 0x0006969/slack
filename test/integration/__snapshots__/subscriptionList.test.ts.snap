// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration: subscription list works 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "fallback": "<#C2147483705> is subscribed to 2 repositories",
      "text": "<https://github.com/atom/atom|atom/atom>
<https://github.com/kubernetes/kubernetes|kubernetes/kubernetes>",
      "title": "<#C2147483705> is subscribed to",
    },
  ],
  "response_type": "in_channel",
}
`;
